File: kashstashandroid/app/src/main/java/com/pulseai/kashstash/RecentTagsManager.kt
package com.pulseai.kashstash

import java.time.Instant

class RecentTagsManager {
    fun updateRecentTags(tags: String, config: KashStashConfig): KashStashConfig {
        if (tags.isBlank()) return config

        val timestamp = Instant.now().toString()
        val recentTags = config.recentTags.toMutableList()

        // Remove existing entry if it exists (we'll re-add with new timestamp)
        recentTags.removeAll { it.value == tags }

        // Add at beginning with new timestamp
        recentTags.add(0, RecentTag(tags, timestamp))

        // Keep only 50 most recent
        val trimmed = recentTags.take(50)

        // Return new config with updated tags
        return config.copy(recentTags = trimmed)
    }

    fun getRecentTagsList(config: KashStashConfig): List<RecentTag> {
        return config.recentTags.take(10)
    }
}

File: Kash Stash/Services/RecentTagsManager.swift
import Foundation

class RecentTagsManager {
    static func addTags(_ tagsString: String, to config: inout AppConfig) {
        let tags = tagsString
            .split(separator: ",")
            .map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
            .filter { !$0.isEmpty }
        
        let now = Date()
        
        for tag in tags {
            // Remove if already exists
            config.recentTags.removeAll { $0.value.lowercased() == tag.lowercased() }
            // Add at front
            config.recentTags.insert(RecentTag(value: tag, lastUsed: now), at: 0)
        }
        
        // Keep only most recent
        if config.recentTags.count > RecentTag.maxRecentTags {
            config.recentTags = Array(config.recentTags.prefix(RecentTag.maxRecentTags))
        }
    }
    
    static func getRecentTagsString(from config: AppConfig, limit: Int = 10) -> [String] {
        Array(config.recentTags.prefix(limit)).map { $0.value }
    }
}
